doc/kernel/code-relocation.rst

```diff
@@ -76,21 +76,30 @@ This section shows additional configuration options that can be set in
 
   .. code-block:: none
 
-     zephyr_code_relocate(src/file1.c SRAM2)
-     zephyr_code_relocate(src/file2.c.c SRAM)
+     zephyr_code_relocate(FILES src/file1.c LOCATION SRAM2)
+     zephyr_code_relocate(FILES src/file2.c LOCATION SRAM)
 
 * if the memory type is appended with _DATA, _TEXT, _RODATA or _BSS, only the
   selected memory is placed in the required memory region.
   for example:
 
   .. code-block:: none
 
-     zephyr_code_relocate(src/file1.c SRAM2_DATA)
-     zephyr_code_relocate(src/file2.c.c SRAM2_TEXT)
+     zephyr_code_relocate(FILES src/file1.c LOCATION SRAM2_DATA)
+     zephyr_code_relocate(FILES src/file2.c LOCATION SRAM2_TEXT)
 
 * Multiple regions can also be appended together such as: SRAM2_DATA_BSS.
   This will place data and bss inside SRAM2.
 
+* Multiple files can be passed to the FILES argument, or CMake generator
+  expressions can be used to relocate a comma-separated list of files
+
+  .. code-block:: none
+
+     file(GLOB sources "file*.c")
+     zephyr_code_relocate(FILES ${sources} LOCATION SRAM)
+     zephyr_code_relocate(FILES $<TARGET_PROPERTY:my_tgt,SOURCES> LOCATION SRAM)
```

Comment:
as described in other comment, then I believe this example will not work unfortunately, because the python relocation script will expect the files to be comma separated,

But let's wait with adjusting this until the final approach has been decided.